R/coercion.r

```diff
@@ -670,10 +670,14 @@ setMethod(f = "as_date", signature = "numeric",
 #' @export
 setMethod("as_date", "character",
           function(x, tz = NULL, format = NULL) {
+            if (!missing(tz)) {
```

Comment:
I would prefer `!is.null` here instead. This one will get the warning even if the supplier passes `NULL` as an argument. 

While unlikely in this case, the caller might want to have the same argument contracts as with `as_date_time` on some programatic occasions.